javascript/code-challenges/challenges-04.test.js

```diff
@@ -0,0 +1,365 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named updateAnimal that takes in array of animals (strings) and some callback function.
+
+This function should first create a new array. Then iterate over the input array and modify each value based on the callback function provided.
+
+Push each updated animal string into the new array. Return the new array.
+
+HINT: Look at the tests to see how the callback functions are used.
+
+------------------------------------------------------------------------------------------------ */
+
+function upper(str) {
+  return str.toUpperCase();
+}
+
+function lower(str) {
+  return str.toLowerCase();
+}
+
+const updateAnimal = (arr, callback) => {
+  let results = [];
+
+  arr.forEach(str => {
+    let result = callback(str);
+    results.push(result);
+  });
+
+  return results;
+};
```

Comment:
You can use .map() here to help you expedite the process.
Since map create a new array and pushes all the data into it for you, you don't have to manually code it all out.
```js
const updateAnimal = (arr, callback) => {
  let results = arr.map(result => callback(result));
  return results;
}
```
or even more simply
```js
const updateAnimal = (arr, callback) => {
  return arr.map(animal => callback(animal));
}
```

Reply:
You could also one-line this.